internal/impl/kafka/input_kafka_franz.go

```diff
@@ -95,6 +96,10 @@ Finally, it's also possible to specify an explicit offset to consume from by add
 			Description("Determines whether to consume from the oldest available offset, otherwise messages are consumed from the latest offset. The setting is applied when creating a new consumer group or the saved offset no longer exists.").
 			Default(true).
 			Advanced()).
+		Field(service.NewBoolField("reconnect_on_unknown_topic").
+			Description("Determines whether to close the client and force a reconnect after seeing an UNKNOWN_TOPIC_OR_PARTITION or UNKNOWN_TOPIC_ID error.").
+			Default(false).
```

Comment:
Let's add a version to this field:

```suggestion
			Version("1.8.0").
			Default(false).
```